build/storage/python_system_tools/test_platform.py

```diff
@@ -0,0 +1,266 @@
+        return False
+
+    @property
+    def system(self) -> Optional[str]:
+        """
+        Checks if kvm modules are loaded
+
+        Returns
+        -------
+        str | None
+            The name of operating system or None if it's unrecognized
+
+        Raises
+        ------
+        UnsupportedOSError
+            The error raised when the operating system is unrecognized (different than Fedora or Ubuntu)
+        """
+
+        if platform.system() == 'Linux':
+            out, _ = self.terminal.execute("cat /etc/os-release")
+            if match := re.search(r"(?<=NAME=)\w+", out):
+                system = match.group()
+            if match is None or system not in PACKAGE_MANAGERS:
+                raise UnsupportedOSError("Unsupported operating system")
+            return system
+
+    @property
+    def package_manager(self):
+        return PACKAGE_MANAGERS[self.system]
+
+    def install(self, program: str) -> bool:
+        """
+        Installs a program on a machine
+
+        Parameters
+        ----------
+        program: str
+            The name of a program to install
+
+        Returns
+        -------
+        bool
+            True if program is installed successfully or present on a machine else False
+        """
+
+        logging.info(f"Installing {program}")
+        out, rc = self.terminal.execute(f"{self.package_manager} -y install {program}", root=True)
+        if rc:
+            logging.error(
+                f"Installing {program} failed with code {rc} and output {out}"
+            )
+            return False
+        logging.info(f"{program} installed successfully or present on a machine")
+        return True
+
+    def setup_docker_compose(self) -> bool:
+        """
+        Checks if docker-compose is installed and installs it if it's not
+
+        Returns
+        -------
+        bool
+            True if setup is successful or docker-compose is already installed else False
+        """
+
+        cmds = (
+            'curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose',
+            "chmod +x /usr/local/bin/docker-compose",
+            "ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose",
+        )
+        _, rc = self.terminal.execute("docker-compose --version", raise_on_error=False)
+        if rc:
+            logging.info("Setting up docker-compose")
+            for cmd in cmds:
+                out, rc = self.terminal.execute(cmd, root=True)
+                if rc:
+                    logging.error(
+                        f"Setting up docker failed with code {rc} and output {out}"
+                    )
+                    return False
+            logging.info(f"Setup of docker-compose was successful")
+            return True
+        logging.info(f"docker-compose is already installed")
+        return True
+
+    def setup_libguestfs_tools(self) -> bool:
```

Comment:
This needed only for KVM host-target platform and not needed in this class

Reply:
Removed